tests/test_wcs_utils.py

```diff
@@ -45,16 +45,54 @@ def test_wcs_to_dict(self):
             self.assertAlmostEqual(new_dict[key], self.header_dict[key])
 
     def test_append_wcs_to_hdu_header(self):
-        pri = fits.PrimaryHDU()
-        self.assertFalse("CRVAL1" in pri.header)
-        self.assertFalse("CRVAL2" in pri.header)
-        self.assertFalse("CRPIX1" in pri.header)
-        self.assertFalse("CRPIX2" in pri.header)
-
-        append_wcs_to_hdu_header(self.wcs, pri.header)
-        for key in self.header_dict:
-            self.assertTrue(key in pri.header)
-            self.assertAlmostEqual(pri.header[key], self.header_dict[key])
+        for use_dictionary in [True, False]:
+            if use_dictionary:
+                wcs_info = self.header_dict
+            else:
+                wcs_info = self.wcs
+
+        # Run the dictionary and full WCS tests as separate subtests
+        with self.subTest(i=use_dictionary):
+            pri = fits.PrimaryHDU()
+            self.assertFalse("CRVAL1" in pri.header)
+            self.assertFalse("CRVAL2" in pri.header)
+            self.assertFalse("CRPIX1" in pri.header)
+            self.assertFalse("CRPIX2" in pri.header)
+
+            append_wcs_to_hdu_header(wcs_info, pri.header)
+            for key in self.header_dict:
+                self.assertTrue(key in pri.header)
+                self.assertAlmostEqual(pri.header[key], self.header_dict[key])
```

Comment:
I'm a bit confused here. I presume the subTests are supposed to be within scope of the the `for use_dictionary in [True, False]` loop?

Reply:
Good catch. It's off that this worked during testing.

Reply:
I fixed this.